nixos/modules/services/monitoring/staytus.nix

```diff
@@ -0,0 +1,370 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.staytus;
+
+  bundler = pkgs.bundler;
+  package = pkgs.staytus;
+  ruby = package.ruby;
+
+  databaseYml = ''
+    production:
+      adapter: mysql2
+      pool: 5
+      database: ${cfg.databaseName}
+      host: ${cfg.databaseHost}
+      password: ${cfg.databasePassword}
+      username: ${cfg.databaseUsername}
+      encoding: utf8
+  '';
+
+  configurationYml = ''
+    default:
+
+      ${cfg.extraConfig}
+  '';
+
+  procfileOptions = ''
+app_name: staytus
+processes:
+  web:
+    restart_mode: usr2
+  worker:
+    restart_mode: start-term
+log_path: /run/staytus/log/staytus.log
+log_root: /run/staytus/log
+pid_root: /run/staytus/pids
+env:
+${concatStrings (mapAttrsToList (name: value: "  ${name}: ${value}\n") staytusEnv)}
+'';
+
+ procfile = ''
+web: ${staytus-rake}/bin/staytus-puma
+worker: ${staytus-rake}/bin/staytus-jobs
+  '';
+
+  unpackTheme = unpack "theme";
+  unpack = id: (name: source:
+    pkgs.stdenv.mkDerivation {
+      name = "staytus-${id}-${name}";
+      buildInputs = [ pkgs.unzip ];
+      buildCommand = ''
+        mkdir -p $out
+        cd $out
+        unpackFile ${source}
+      '';
+  });
```

Comment:
Are staytus themes always zip archives? The following api might be more general:

```
services.staytus.themes = [
  (fetchzip {
    url = "https://host/theme.zip";
    sha256 = "...";
  })
  (fetchgit {
    url = "https://host/theme.git"
    rev = "some-release";
    sha256 = "...";
  })
];
```

Reply:
That would be much better, yes. Could you share an example of how to handle that at the receiving end? I just use unpackFile at the moment.

Reply:
@Mic92 actually in this api, would it be possible to use local files as well as remote ones? Is there a local version of `fetchzip`, or is it compatible with relative paths?

Reason is, I am using this in nixops, and have my company theme in the same dir as the server nixops config, so I just use ./theme.zip.